esp/bindings/http/platform/httpbinding.cpp

```diff
@@ -763,18 +763,23 @@ static void filterXmlBySchema(IPTree* in, IXmlType* type, const char* tag, Strin
     }
 }
 
-static void filterXmlBySchema(StringBuffer& in, StringBuffer& schema, StringBuffer& out)
+static void filterXmlBySchema(StringBuffer& in, StringBuffer& schema, const char* qName, StringBuffer& out)
 {
     Owned<IXmlSchema> sp = createXmlSchema(schema);
     Owned<IPTree> tree = createPTreeFromXMLString(in);
 
     //VStringBuffer name("tns:%s", tree->queryName());
-    const char* name = tree->queryName();
+    const char* name = qName;
     IXmlType* type = sp->queryElementType(name);
     if (!type)
     {
-        StringBuffer method(strlen(name)-7, name);
-        type = sp->queryElementType(method);
+        name = tree->queryName();
+        type = sp->queryElementType(name);
+        if (!type)
+        {
+            StringBuffer method(strlen(name)-7, name);
```

Comment:
Is it safe to assume that name (tree->queryName()) is over 7 chars and I guess that assumes it ends in "Request"?